source/s3_meta_request.c

```diff
@@ -1592,13 +1593,16 @@ void aws_s3_meta_request_finish_default(struct aws_s3_meta_request *meta_request
 struct aws_future_bool *aws_s3_meta_request_read_body(
     struct aws_s3_meta_request *meta_request,
     struct aws_byte_buf *buffer) {
-
+    __itt_task_begin(s3_domain, __itt_null, __itt_null, __itt_string_handle_create("read"));
     AWS_PRECONDITION(meta_request);
     AWS_PRECONDITION(buffer);
 
     /* If async-stream, simply call read_to_fill() */
     if (meta_request->request_body_async_stream != NULL) {
-        return aws_async_input_stream_read_to_fill(meta_request->request_body_async_stream, buffer);
+        struct aws_future_bool *result =
+            aws_async_input_stream_read_to_fill(meta_request->request_body_async_stream, buffer);
+        __itt_task_end(s3_domain);
```

Comment:
we should probably mark up the implementation of `aws_input_stream_read()`, begin/end task right around the vtable call

it wil be a lot harder to mark up the async stream :|